projects/blenderbot2/agents/README.md

```diff
@@ -0,0 +1,118 @@
+# BlenderBot 2 Agent Code: FAQ
+
+Below, we have compiled a list of FAQs regarding the usage of BlenderBot2. Please open an issue if your issue is not addressed below.
+
+### `ModuleNotFoundError: No module named 'transformers'`
+
+Please run `pip install transformer==4.3.3`.
+
+### `ModuleNotFoundError: No module named 'parlai.zoo.bart.bart_large`
+Please make sure you have [installed fairseq](https://github.com/pytorch/fairseq#requirements-and-installation)
+
+### `ModuleNotFoundError: No module named 'parlai.zoo.blenderbot2'`
+If you have installed ParlAI from source, make sure you have pulled from the main branch. If you have installed via pip, make sure you are on version 1.4.0 or later.
+
+### `ValueError: Must provide a valid server for search`
+You'll need to setup your own search server; see discussion in [#3816](https://github.com/facebookresearch/ParlAI/issues/3816)
+
+### `AssertionError`
+```
+    assert search_queries
+AssertionError
+```
+
+Consult "How can I use BlenderBot2 with **only** search" below.
+
+### `IndexError`
+```
+    File "/home/ParlAI/projects/blenderbot2/agents/modules.py", line 556, in <dictcomp>
+    batch_id: memory_vec[batch_id, : num_memories[mem_id]]
+IndexError: index 3 is out of bounds for dimension 0 with size 3
+```
+Make sure you're providing memories appropriately to the model; the field in which these are emitted is specified by the `--memory-key`. If you want to extract memories from the full context, set `--memory-key full_text`
+
+```
+    File "/home/ParlAI/projects/blenderbot2/agents/modules.py", line 556, in <dictcomp>
+    for batch_id, mem_id in enumerate(indices)
+IndexError: index 1 is out of bounds for dimension 0 with size 1
+```
+If you notice this during training, please set `--memory-decoder-model-file ''`.
+
+### How can I use BlenderBot2 with **only** search?
+
+You'll need to do two things:
+
+1. Set `--knowledge-access-method search_only`
+2. Set `--query-generator-model-file zoo:sea/bart_sq_gen/model`
```

Comment:
Is this one necessary? Isn't that the default atm?

Reply:
No, this is not the default. The default uses `--knowledge-access-method classify`, with the query generator from BB2 (trained to either search or retrieve from memory)

Reply:
Sorry, I was referring to `--query-generator-model-file zoo:sea/bart_sq_gen/model`. That can be default, right?

Reply:
At the moment it's not, the default is `zoo:blenderbot2/query_generator/model`, which determines whether to search or retrieve from memory

Reply:
Oh, yeah, I remember now.